src/ui/ResultLink/ResultLink.ts

```diff
@@ -478,11 +478,10 @@ export class ResultLink extends Component {
   );
 
   private filterProtocol(uri: string) {
-    if (/^(https?|ftp|file|mailto|tel):/i.test(uri)) {
-      return uri;
-    }
+    const isAbsolute = /^(https?|ftp|file|mailto|tel):/i.test(uri);
```

Comment:
Just wondering if there was anything we could have done just using the browser API to "try and build this link please", and then get the attributes from the the URL object instead of these regex ?

https://developer.mozilla.org/en-US/docs/Web/API/URL

Reply:
I liked this idea and explored the approach, but I found two limitations.

1. When the url starts with `mailto`, the parser cannot parse.
2. For relative urls, the parser is not very strict. I suspect it is possible to bypass. For example, in the screenshot below, the parser finds a `pathname` even though the relative url (first arg to `new URL`) is not a valid pathname. We would still need to check the string,

![XSS_using_relative_url](https://user-images.githubusercontent.com/5565841/99459069-8056b300-28fb-11eb-9b99-fa53c99663d8.PNG)